unit-tests-ext/src/test/scala/java/nio/file/attribute/FileTimeExtTest.scala

```diff
@@ -0,0 +1,106 @@
+package java.nio.file.attribute
+
+import java.time.Instant
+import java.util.concurrent.TimeUnit
+import org.junit.Test
+import org.junit.Assert._
+
+class FileTimeExtTest {
+  private val timestamp          = 1582230020000L
+  private val timestampAsSeconds = timestamp / 1000L
+  private val timestampAsNanos   = timestamp * 1e6.toLong
+  private val nanosAdjustment    = 1234L
+
+  @Test def fromInstant(): Unit = {
+    def fileTimeFromInstantEquals(ft: FileTime, instant: Instant) = {
+      assertEquals(timestamp, instant.toEpochMilli)
+      assertEquals(timestamp, ft.toMillis)
+
+      val fromInstant = FileTime.from(instant)
+      assertEquals(0, ft.compareTo(fromInstant))
+      assertEquals(timestamp, fromInstant.toMillis)
+      assertEquals(timestampAsSeconds, fromInstant.to(TimeUnit.SECONDS))
+      assertEquals(ft, fromInstant)
+    }
+
+    fileTimeFromInstantEquals(FileTime.fromMillis(timestamp),
+                              Instant.ofEpochMilli(timestamp))
+
+    fileTimeFromInstantEquals(
+      FileTime.from(timestampAsSeconds, TimeUnit.SECONDS),
+      Instant.ofEpochSecond(timestampAsSeconds))
+
+    fileTimeFromInstantEquals(
+      FileTime.from(timestampAsNanos + nanosAdjustment, TimeUnit.NANOSECONDS),
+      Instant.ofEpochSecond(timestampAsSeconds, nanosAdjustment))
+  }
+
+  @Test def fromInstantMaxPrecision(): Unit = {
+    val maxSecondsWithNanos  = Long.MaxValue / 1e9.toLong - 1L
+    val naxSecondsWithMicros = Long.MaxValue / 1e6.toLong - 1L
+    val maxSecondsWithMillis = Long.MaxValue / 1e3.toLong - 1L
+
+    def toUnit(seconds: Long, nanos: Long, unit: TimeUnit) = {
+      val fromSeconds = unit.convert(seconds, TimeUnit.SECONDS)
+      val fromNanos   = unit.convert(nanos, TimeUnit.NANOSECONDS)
+      fromSeconds + fromNanos
+    }
+
+    def fromSeconds(seconds: Long, unit: TimeUnit) =
+      FileTime.from(toUnit(seconds, Instant.MAX.getNano, unit), unit)
```

Comment:
What does `Instant.MAX.getNano` represent? Why is this better than using `0L`? This deserves a comment.